### PR TITLE
fix: System messages on the user being removed/leaving the conversation or team

### PR DESCRIPTION
**Pull request by makingthematrix**
fixes https://wearezeta.atlassian.net/browse/AN-6859
fixes https://wearezeta.atlassian.net/browse/AN-6909

In the past, when a user was removed from a conversation, a `MemberLeaveEvent` should have come and only in the response to that event we should have displayed a system message "User left" or "User was removed" or "You removed user" or "Another user removed user". Now the situation became more complex: We can no longer rely on events and a user may be removed
from the conversation because they were removed from the team entirely (which we can discover either because of another event type, `MemberEvent.MemberLeave` or in a few different ways.

This PR ensures that we display system messages when we discover that the user left or was
removed from the conversation in any of those ways and also tries to find out if we don't
do it more than once.

#### APK
[Download build #2139](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2139/artifact/build/artifact/wire-dev-PR2859-2139.apk)
[Download build #2157](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2157/artifact/build/artifact/wire-dev-PR2859-2157.apk)

### CODE REVIEW

**Comment by makingthematrix** [comment]
build again

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/MessagesStorage.scala [hunk 1]
```diff
@@ -64,6 +64,7 @@ trait MessagesStorage extends CachedStorage[MessageId, MessageData] {
   //System message events no longer have IDs, so we need to search by type, timestamp and sender
   def hasSystemMessage(conv: ConvId, serverTime: RemoteInstant, tpe: Message.Type, sender: UserId): Future[Boolean]
 
+  def getLastSystemMessage(conv: ConvId, tpe: Message.Type, sender: UserId): Future[Option[MessageData]]
```
Comment (gizemb) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/MessagesStorage.scala:67]: can we add tests for that?

Comment (makingthematrix) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/MessagesStorage.scala:67]: @gizemb : Not really. There's no easy way to mock the underlying storage. Usually I try to keep methods in the `*Storage` classes as simple as possible and mock them to test methods in the services.

Comment (android10) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/MessagesStorage.scala:67]: Why is this happening? Current architecture constraints does not let us to mock the Storage?

Comment (makingthematrix) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/MessagesStorage.scala:67]: @android10 : I used a wrong word. We can't mock what lies below storage. Or at least it's so complicated that it's not worth to do it to test very simple methods in `*Storage`. In this case, for example, `getLastSystemMessage` is only two lines long and to test we would have to mock a much more complex `CachedStorage.find`.

**Pull request merged**